docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+     -H "Signature: MEYCIQDwogJW4wuIBGO5MYVLt1F0nFq+6BsULpKpmCcyGrcx/QIhAMj1uhj6LLW14pF6zZGToP5rdd0tgx9fvwUDHLXLbDO+" \
+     -X GET http://127.0.0.1:6001/provenance/data/history/db2/key3 | jq .
+```
+
+**Output**
+
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "values": [
+      {
+        "value": "dGhpcyBpcyBhIHRoaXJkIHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEQCIFE2JIa9MkaGtBEZUqx8MIsYsSSiJS0Atz+TkI6Yl73HAiBRU98GTF+FlQoVXvSuqZArwd5UxjK5aMXBJLEdRCQKTA=="
+}
+```
+
+Key `key3` changed only once, in block 5
+
+> Please note that due `json.Marshal` for protobuf objects limitations, in many cases inside `version` json object, only `block_num` field is presented, without `tx_num`, because `tx_num` equal to zero. But during unmarshalling protobuf object back from json, `tx_num` field filed will be assigned zero value.
+
+### History query with _only_deletes_ option
+As you can see, last query returned value for `<db2, key3>`. Now we want to query if key was deleted at some time, and their last value before delete. We can run same query, but only with `only_deletes` turned `true`
+
+**Sign json query data**
+```sh
+bin/signer -data '{"user_id":"alice","db_name":"db2","key":"key3","only_deletes":true}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Output**
+```
+MEUCIQCPTVwKrosS6+2WnOUJ4qBU8Ru7ubJYkTOljCXOkFPAtgIgcFftqCm7bOZLxLkfzWloIznqpOkoTegyYtX+xgfHd6s=
+```
+
+Submit query
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEUCIQCPTVwKrosS6+2WnOUJ4qBU8Ru7ubJYkTOljCXOkFPAtgIgcFftqCm7bOZLxLkfzWloIznqpOkoTegyYtX+xgfHd6s=" \
+     -X GET http://127.0.0.1:6001/provenance/data/history/db2/key3?onlydeletes=true | jq .
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "values": [
+      {
+        "value": "dGhpcyBpcyBhIHRoaXJkIHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEUCIEk90HGFa90Yr/dBDONE+GVD4GaowdQFRL25+S0ECY7tAiEA7+gJSZD8pt0v+KC318xxyLTmLrzcBPnhEMIb/xUmJtI="
+}
+```
+As we can see here, the value returned by query is last value before key was deleted. If we run same query for `key2`, we will get empty list.
+
+### History query for specific version
+Let's check what was value of `key1` at time then tx with position `{"block_num": 6, "tx_num": 0}` was committed.  
```

Comment:
Let's check what was value of `key1` at time then tx with position `{"block_num": 6, "tx_num": 0}` was committed.
->

Let's check what was the value of `key1` at the time when tx with version `{"block_num": 6, "tx_num": 0}` was committed.

Reply:
done